CI: ci-ipsec-e2e: nil pointer in xDS  stream handling

## CI failure

Test output

```
[=] [cilium-test-1] Test [check-log-errors] [10/10]
.....
  [-] Scenario [check-log-errors/no-errors-in-logs]
  [.] Action [check-log-errors/no-errors-in-logs:kind-kind/kube-system/cilium-kfznq (apply-sysctl-overwrites)]
  [.] Action [check-log-errors/no-errors-in-logs:kind-kind/kube-system/cilium-kfznq (mount-bpf-fs)]
  [.] Action [check-log-errors/no-errors-in-logs:kind-kind/kube-system/cilium-kfznq (clean-cilium-state)]
  [.] Action [check-log-errors/no-errors-in-logs:kind-kind/kube-system/cilium-kfznq (install-cni-binaries)]
  [.] Action [check-log-errors/no-errors-in-logs:kind-kind/kube-system/cilium-kfznq (cilium-agent)]
  ❌ Non-zero (1) restart count of kind-kind/kube-system/cilium-kfznq (cilium-agent) must be investigated
...
Test [check-log-errors]:
  🟥 check-log-errors/no-errors-in-logs:kind-kind/kube-system/cilium-kfznq (cilium-agent): Found 1 logs in kind-kind/kube-system/cilium-kfznq (cilium-agent) from before pod restart matching list of errors that must be investigated:
panic: runtime error: invalid memory address or nil pointer dereference (1 occurrences)
```

Previous Pod log of restarted agent from sysdump

```
2026-01-07T12:56:40.034106478Z time=2026-01-07T12:56:40.033462114Z level=debug source=/go/src/github.com/cilium/cilium/pkg/envoy/xds/server.go:431 msg="starting watch resources" module=agent.controlplane.envoy-proxy xdsStreamID=6 xdsClientNode=host~127.0.0.1~no-id~localdomain version=8 xdsTypeURL=type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret xdsNonce=8 resources=1
2026-01-07T12:56:40.034113713Z time=2026-01-07T12:56:40.03353778Z level=debug source=/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:204 msg="EventQueue event processing statistics" endpointID=52 ipv6=fd00:10:244:1::6560 containerID=bbed203841 desiredPolicyRevision=0 datapathPolicyRevision=0 ipv4=10.244.1.61 containerInterface="" k8sPodName=cilium-test-1/client3-74887fc475-m6jxf ciliumEndpointName=cilium-test-1/client3-74887fc475-m6jxf subsys=endpoint name=endpoint-52 eventType=*endpoint.EndpointRegenerationEvent eventHandlingDuration=21.243404ms eventEnqueueWaitTime=1.079µs eventConsumeOffQueueWaitTime=90.733µs
2026-01-07T12:56:40.034116439Z time=2026-01-07T12:56:40.033638207Z level=debug source=/go/src/github.com/cilium/cilium/pkg/envoy/xds/watcher.go:169 msg="context canceled, terminating resource watch" module=agent.controlplane.envoy-proxy xdsAckedVersion=8 xdsClientNode=127.0.0.1 xdsTypeURL=type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret
2026-01-07T12:56:40.034983003Z time=2026-01-07T12:56:40.034885277Z level=debug source=/go/src/github.com/cilium/cilium/pkg/envoy/xds/server.go:181 msg="stopping xDS stream handling" module=agent.controlplane.envoy-proxy xdsStreamID=6 xdsClientNode=host~127.0.0.1~no-id~localdomain
2026-01-07T12:56:40.036652182Z panic: runtime error: invalid memory address or nil pointer dereference
2026-01-07T12:56:40.036874120Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x29ff12b]
2026-01-07T12:56:40.036882431Z 
2026-01-07T12:56:40.036950023Z goroutine 4944 [running]:
2026-01-07T12:56:40.036954660Z github.com/cilium/cilium/pkg/envoy/xds.(*Server).processRequestStream(0xc000a63a70, {0x51cbdb8, 0xc005d6bf50}, 0xc000c69870, {0x7f3019f06db8, 0xc000c69860}, 0xc005cebdc0, {0x4bc4cca, 0x44}, {0x0, ...})
2026-01-07T12:56:40.036958360Z 	/go/src/github.com/cilium/cilium/pkg/envoy/xds/server.go:440 +0x17cb
2026-01-07T12:56:40.036961092Z github.com/cilium/cilium/pkg/envoy/xds.(*Server).HandleRequestStream(0xc000a63a70, {0x51cbdb8, 0xc005d6bf50}, {0x7f3019f06db8, 0xc000c69860}, {0x4bc4cca, 0x44}, {0x0, 0x0})
2026-01-07T12:56:40.036963726Z 	/go/src/github.com/cilium/cilium/pkg/envoy/xds/server.go:188 +0x2d1
2026-01-07T12:56:40.036997606Z github.com/cilium/cilium/pkg/envoy.(*xdsGRPCServer).StreamSecrets(0xc000a63a70, {0x51fa988, 0xc000c69860})
2026-01-07T12:56:40.037003037Z 	/go/src/github.com/cilium/cilium/pkg/envoy/grpc.go:166 +0x74
2026-01-07T12:56:40.037005808Z github.com/envoyproxy/go-control-plane/envoy/service/secret/v3._SecretDiscoveryService_StreamSecrets_Handler({0x4930e60?, 0xc000a63a70}, {0x51e8ac8, 0xc006763100})
2026-01-07T12:56:40.037009894Z 	/go/src/github.com/cilium/cilium/vendor/github.com/envoyproxy/go-control-plane/envoy/service/secret/v3/sds_grpc.pb.go:177 +0xd8
2026-01-07T12:56:40.037012899Z google.golang.org/grpc.(*Server).processStreamingRPC(0xc0019f46c8, {0x51cbdb8, 0xc005d6bef0}, 0xc000ed0b60, 0xc000a63b30, 0x80828c0, 0x0)
2026-01-07T12:56:40.037015543Z 	/go/src/github.com/cilium/cilium/vendor/google.golang.org/grpc/server.go:1714 +0x1172
2026-01-07T12:56:40.037017678Z google.golang.org/grpc.(*Server).handleStream(0xc0019f46c8, {0x51ce418, 0xc001efd380}, 0xc000ed0b60)
2026-01-07T12:56:40.037019965Z 	/go/src/github.com/cilium/cilium/vendor/google.golang.org/grpc/server.go:1836 +0xd85
2026-01-07T12:56:40.037022790Z google.golang.org/grpc.(*Server).serveStreams.func2.1()
2026-01-07T12:56:40.037025479Z 	/go/src/github.com/cilium/cilium/vendor/google.golang.org/grpc/server.go:1063 +0x7f
2026-01-07T12:56:40.037030044Z created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 2679
2026-01-07T12:56:40.037032376Z 	/go/src/github.com/cilium/cilium/vendor/google.golang.org/grpc/server.go:1074 +0x11d
```

CI run: https://github.com/cilium/cilium/actions/runs/20782022548/job/59681632887

--- a/pkg/envoy/xds/server.go
+++ b/pkg/envoy/xds/server.go
@@ -437,8 +437,10 @@ func (s *Server) processRequestStream(ctx context.Context, streamLog *slog.Logge
 
 		default: // Pending watch response.
 			state := &typeStates[chosen]
-			state.pendingWatchCancel()
-			state.pendingWatchCancel = nil
+			if state.pendingWatchCancel != nil {
+				state.pendingWatchCancel()
+				state.pendingWatchCancel = nil
+			}
 
 			if !recvOK {
 				// chosen channel was closed. If context has an error (e.g.,

